src/coreclr/tools/Common/TypeSystem/MetadataEmitter/TypeSystemMetadataEmitter.cs

```diff
@@ -424,10 +424,22 @@ public void EmitArrayShapeAtCurrentIndexStack(BlobBuilder signatureBuilder, int
 
                 if (!emittedWithShape)
                 {
-                    shapeEncoder.Shape(rank, ImmutableArray<int>.Empty, ImmutableArray<int>.Empty);
+                    shapeEncoder.Shape(rank, ImmutableArray<int>.Empty, ImmutableArraysFilledWithZeroes[rank]);
                 }
             }
 
+            static ImmutableArray<int>[] ImmutableArraysFilledWithZeroes = CreateStaticArrayOfImmutableArraysFilledWithZeroes(33); // The max rank of an array is 32
```

Comment:
If this can come from user code, might not hurt to add a fallback that just creates a fresh immutable array if rank is more than 32. C# has no problem emitting such array.

Reply:
Sure, seems reasonable.